app/helpers/password_resets_helper.rb

```diff
@@ -0,0 +1,24 @@
+module PasswordResetsHelper
+  def extract_user_from_reset_params
+    @user = User.find_by(email: params[:password_reset][:email].downcase)
+  end
+
+  def get_user_by_email
+    @user = User.find_by(email: params[:email].downcase)
+  end
+
+  def valid_user
+    unless @user && @user.activated? && @user.authenticated?(:reset, params[:id])
+      redirect_to root_url, flash: {
```

Comment:
What's up with the `{}` around flash messages?

Reply:
So when I use
```suggestion
      redirect_to root_url, notice: 'message to display'
``` it does not display the message unless I put them in a containing hash `flash`, the other option is using `flash[:notice]` before render/redirect, which you have discouraged. The reason the former doen't work is because of how I have used a `flash.each` in the `application.html.erb` file to display all error messages instead of having specific ones for each page